activemodel/lib/active_model/secure_password.rb

```diff
@@ -120,18 +122,51 @@ module ClassMethods
       #
       #   # raises ActiveSupport::MessageVerifier::InvalidSignature since the token is expired
       #   User.find_by_password_reset_token!(token)
-      def has_secure_password(attribute = :password, validations: true, reset_token: true)
+      #
+      # ===== Customizing the hashing algorithm
+      # To add a custom algorithm, create a class that implements +hash_password+, +verify_password+, +password_salt+,
+      # +validate+ and +algorithm_name+ methods.
+      #
+      #   class Argon2Password
+      #     def initialize
+      #       require "argon2"
+      #     end
+      #
+      #     def hash_password(unencrypted_password)
+      #       Argon2::Password.create(unencrypted_password)
+      #     end
+      #
+      #     def verify_password(password, digest)
+      #       Argon2::Password.verify_password(password, digest)
+      #     end
+      #
+      #     def password_salt(digest)
+      #       Argon2::HashFormat.new(digest).salt
+      #     end
+      #
+      #     def validate(_record, _attribute)
+      #       # Argon2 has no maximum input size, no validation needed
```

Comment:
Although Argon2 apparently doesn't need any validation, it would be nice to show how a validation should be implemented.